Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs

```diff
@@ -46,8 +96,45 @@ public static MSBuildToolsConfig Load()
                 EditorJsonUtility.FromJsonOverwrite(File.ReadAllText(MSBuildSettingsFilePath), toReturn);
             }
 
+            bool needToSave = false;
+
+            toReturn.DependenciesProjectGuid = EnsureGuid(ref toReturn.dependenciesProjectGuid, ref needToSave);
+
+            toReturn.AssemblyCSharpGuid = EnsureGuid(ref toReturn.assemblyCSharpGuid, ref needToSave);
+            toReturn.AssemblyCSharpEditorGuid = EnsureGuid(ref toReturn.assemblyCSharpEditorGuid, ref needToSave);
+            toReturn.AssemblyCSharpFirstPassGuid = EnsureGuid(ref toReturn.assemblyCSharpFirstPassGuid, ref needToSave);
+            toReturn.AssemblyCSharpFirstPassEditorGuid = EnsureGuid(ref toReturn.assemblyCSharpFirstPassEditorGuid, ref needToSave);
+
+            toReturn.BuiltInPackagesFolderGuid = EnsureGuid(ref toReturn.builtInPackagesFolderGuid, ref needToSave);
+            toReturn.ImportedPackagesFolderGuid = EnsureGuid(ref toReturn.importedPackagesFolderGuid, ref needToSave);
+            toReturn.ExternalPackagesFolderGuid = EnsureGuid(ref toReturn.externalPackagesFolderGuid, ref needToSave);
+
+            toReturn.SolutionGuid = EnsureGuid(ref toReturn.solutionGuid, ref needToSave);
+
+            if (CurrentConfigVersion > toReturn.version)
+            {
+                toReturn.version = CurrentConfigVersion;
+                needToSave = true;
+            }
+
+            if (needToSave)
+            {
+                toReturn.Save();
+            }
+
             return toReturn;
         }
+
+        private static Guid EnsureGuid(ref string field, ref bool needToSave)
```

Comment:
>needToSave [](start = 66, length = 10)

Is needToSave supposed to be getting set to 'true' inside the 'if' block?

Reply:
Yes! Dang, good catch.

---
In reply to: [364907055](https://github.com/microsoft/MSBuildForUnity/pull/96#discussion_r364907055) [](ancestors = 364907055)